pkg/apis/controller/v1alpha1/bluegreendeployment_types.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BlueGreenDeploymentSpec defines the desired state of BlueGreenDeployment.
+type BlueGreenDeploymentSpec struct {
+	// Replicas is the number of desired replicas. It determines number of pods for both
+	// Blue and Green ReplicaSets. It defaults to 1 if unspecified.
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// PodSpec defines the specification of the desired behavior of the pods maintained
+	// by the Blue and Green ReplicaSets, such as what image to run.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	PodSpec v1.PodSpec `json:"podSpec,omitempty"`
+}
+
+// BlueGreenDeploymentStatus defines the observed state of BlueGreenDeployment
+type BlueGreenDeploymentStatus struct {
+	// ActiveReplicaSetColor records color of current active ReplicaSet, which is used to
+	// change label selectors of service. It controls which ReplicaSet the service will
+	// send traffic to. It defaults to "blue" color at the beginning.
+	// +optional
+	ActiveReplicaSetColor string `json:"activeReplicaSetColor,omitempty"`
```

Comment:
Would crd schema validation with an enum work for this?

Reply:
As there are two colors, we can definitely use an enum variable to represent current active color. Something like `IsActiveColorBlue bool` will work.